lib/convection/model/template.rb

```diff
@@ -183,9 +197,122 @@ def diff(other, stack_ = nil)
       end
 
       def to_json(stack_ = nil, pretty = false)
+        rendered_stack = render(stack_)
+        validate(rendered_stack)
+
         return JSON.generate(render(stack_)) unless pretty
         JSON.pretty_generate(render(stack_))
       end
+
+      def validate(rendered_stack = nil)
+        validate_resources(rendered_stack)
+        validate_mappings(rendered_stack)
+        validate_parameters(rendered_stack)
+        validate_outputs(rendered_stack)
+        validate_description(rendered_stack)
+        validate_bytesize(rendered_stack)
+
+      end
+
+      def validate_resources(rendered_stack = nil)
+        resources=rendered_stack["Resources"]
+        number_of_resources = resources.count
+        if number_of_resources > CF_MAX_RESOURCES
+          raise ArgumentError, "Error: Excessive Number of Resources (#{number_of_resources}) Max= #{CF_MAX_RESOURCES}"
+        end
+        largest_resource_name = resources.keys.max
```

Comment:
I think you're trying to find the longest resource name in the array?  I don't think that 'max' will do it.

see: http://apidock.com/ruby/Enumerable/max_by